src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -0,0 +1,165 @@
+# lm-buddy entrypoint to run evaluations using a Prometheus inference server
+# see https://github.com/kaistAI/prometheus/blob/main/evaluation/benchmark/run_absolute_scoring.py
+
+import copy
+import json
+from pathlib import Path
+
+from datasets import Dataset, load_dataset
+from fastchat.conversation import get_conv_template
+from openai import Completion, OpenAI, OpenAIError
+from tqdm import tqdm
+from transformers import PreTrainedTokenizer
+
+from lm_buddy.integrations.huggingface import HuggingFaceAssetLoader
+from lm_buddy.integrations.huggingface.tokenizer_config import AutoTokenizerConfig
+from lm_buddy.integrations.wandb import (
+    ArtifactLoader,
+    ArtifactType,
+    build_directory_artifact,
+    wandb_init_from_config,
+)
+from lm_buddy.jobs.common import LMBuddyJobType
+from lm_buddy.jobs.configs import PrometheusJobConfig
+
+
+class BadResponseError(Exception):
+    def __init__(self, message, error=None):
+        self.message = message
+        self.error = error
+
+
+def openai_completion(config: PrometheusJobConfig, client: OpenAI, prompt: str) -> Completion:
+    """Connects to a remote OpenAI-API-compatible Prometheus endpoint
+    and returns a Completion holding the model's response.
+    """
+
+    return client.completions.create(
+        model=config.prometheus.inference.engine,
+        prompt=prompt,
+        best_of=config.prometheus.best_of,
+        max_tokens=config.prometheus.max_tokens,
+        frequency_penalty=config.prometheus.frequency_penalty,
+        temperature=config.prometheus.temperature,
+        top_p=config.prometheus.top_p,
+    )
+
+
+def parse_response(config: PrometheusJobConfig, response: Completion) -> tuple[str, str]:
+    """Given a Prometheus eval response as returned by the OpenAI API
+    endpoint (i.e. in Completion format), extract feedback
+    and score.
+    """
+
+    if response is None:
+        raise BadResponseError("Server returned an empty response")
+
+    try:
+        response_text = response.choices[0].text
+        # note: this can raise a ValueError if the message is malformed
+        feedback, score = response_text.split("[RESULT]")
+        feedback = feedback.strip()
+        score = score.strip()
+        if score not in [
+            str(s) for s in range(config.evaluation.min_score, config.evaluation.max_score + 1)
+        ]:
+            raise BadResponseError(f"Score {score} is not in range")
+    except (ValueError, BadResponseError) as e:
+        raise BadResponseError(f"Server returned a malformed response ({e})", e)
+
+    return feedback, score
+
+
+def instruction_to_prompt(instruction: str) -> str:
+    """Given some text containing Prometheus instructions, transform it
+    into an actual llama-2 prompt.
+    """
+    conv = get_conv_template("llama-2")
```

Comment:
I see your point... In theory yes, it would be cleaner than having something hard-coded in our entrypoint.
In practice, I think that might be overkill as we are developing for a model that has been trained to work in a very specific way and we cannot forsee if these things will change or not in future versions (or if anything else will).
As the extra effort is low I will make these configurable anyway, but I will add defaults in the config class so they do not have to be explicitly specified.